packages/wrangler/src/api/startDevWorker/BundlerController.ts

```diff
@@ -264,16 +265,23 @@ export class BundlerController extends Controller<BundlerControllerEventMap> {
 	async #ensureWatchingAssets(config: StartDevWorkerOptions) {
 		await this.#assetsWatcher?.close();
 
+		const debouncedRefreshBundle = debounce(() => {
+			if (this.#currentBundle) {
+				this.emitBundleCompleteEvent(config, this.#currentBundle);
+			}
+		});
+
 		if (config.assets?.directory) {
 			this.#assetsWatcher = watch(config.assets.directory, {
 				persistent: true,
 				ignoreInitial: true,
 			}).on("all", async (eventName, filePath) => {
 				const message = getAssetChangeMessage(eventName, filePath);
-				logger.debug(`🌀 ${message}...`);
-				if (this.#currentBundle) {
-					this.emitBundleCompleteEvent(config, this.#currentBundle);
-				}
+				logger.log(`🌀 ${message}...`);
+				debouncedRefreshBundle();
+				// if (this.#currentBundle) {
+				// 	this.emitBundleCompleteEvent(config, this.#currentBundle);
+				// }
```

Comment:
```suggestion
```

Reply:
Ah, this was me validating that the test failed without the fix—forgot to remove 🤦